test/test_judiWeights.jl

```diff
@@ -0,0 +1,79 @@
+# Unit tests for judiWeights
+# Rafael Orozco (rorozco@gatech.edu)
+# July 2021
+
+using JUDI.TimeModeling, Test, LinearAlgebra
+import LinearAlgebra.BLAS.axpy!
+
+
+# number of sources/receivers
+nsrc = 1
+nrec = 120
+weight_size_x = 4
+weight_size_y = 4
+################################################# test constructors ####################################################
+
+@testset "judiWeights Unit Tests" begin
+
+    # set up judiWeights
+
+
+    # Extended source weights
+    weights = Array{Array}(undef, nsrc)
+    for j=1:nsrc
+        weights[j] = randn(Float32, weight_size_x,weight_size_y) #model size (100,100)
+    end
+    w = judiWeights(weights)
+
+    @test isequal(w.nsrc, nsrc)
+    @test isequal(typeof(w.weights), Array{Array, 1})
+    @test isequal(size(w), (weight_size_x*weight_size_y,1))
+
+
+
+#################################################### test operations ###################################################
+
+    # conj, transpose, adjoint
+    @test isequal(size(w), size(conj(w)))
+
+    @test isequal(reverse(size(w)), size(transpose(w)))
+
+    @test isequal(reverse(size(w)), size(adjoint(w)))
+
+    # +, -, *, /
+    @test iszero(norm(2*w - (w + w)))
+    @test iszero(norm(w - (w + w)/2))
+
+    #test unary operator
+    @test iszero(norm((-w) - (-1*w))) 
+
+    # vcat
+    w_vcat = [w; w]
+    @test isequal(length(w_vcat), 2*length(w))
+    @test isequal(w_vcat.nsrc, 2*w.nsrc)
+
+    # dot, norm, abs
+    @test isapprox(norm(w), sqrt(dot(w, w)))
+    @test isapprox(abs.(w.weights[1]), abs(w).weights[1]) 
+    @test isapprox(abs.(w), vec(vcat(abs(w).weights...))) #this tests the iterator
+
+    # vector space axioms
+    u = judiWeights(randn(Float32, nsrc))
+    v = judiWeights(randn(Float32, nsrc))
+    w = judiWeights(randn(Float32, nsrc))
+    a = randn(1)[1]
+    b = randn(1)[1]
```

Comment:
`randn(Float32, 1)[1]` can have some bad type conversion inaccuracy in some cases without single precision.